graphql_compiler/compiler/match_query.py

```diff
@@ -141,16 +143,27 @@ def _split_match_steps_into_match_traversals(match_steps):
 
 def convert_to_match_query(ir_blocks):
     """Convert the list of IR blocks into a MatchQuery object, for easier manipulation."""
-    output_block = ir_blocks[-1]
+    where_block = ir_blocks[-1]
+    if not isinstance(where_block, SelectWhereFilter):
+        raise AssertionError(u'Expected last IR block to be SelectWhereFilter, found: '
+                             u'{} {}'.format(where_block, ir_blocks))
+    ir_except_where = ir_blocks[:-1]
+
+    output_block = ir_except_where[-1]
     if not isinstance(output_block, ConstructResult):
         raise AssertionError(u'Expected last IR block to be ConstructResult, found: '
```

Comment:
This isn't the last IR block any more -- the message should be updated if we stick with this representation.